UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.IntegrationTests/RadioNavigationalWarning/Admin/RadioNavigationalWarningAdminTest.cs

```diff
@@ -0,0 +1,167 @@
+            Assert.AreEqual(2, adminListFilter.CurrentPageIndex);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithInValidPageNo_ThenReturnEmptyListAsync()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 3;
+            IActionResult result = await _controller.Index(4, null, null);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(0, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithValidAdminListRecordPerPage_ThenReturnValidRecordPerPageAsync()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 5;
+            IActionResult result = await _controller.Index(1, null, null);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(5, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+            Assert.AreEqual(2, adminListFilter.PageCount);
+            Assert.AreEqual(0, adminListFilter.SrNo);
+            Assert.AreEqual(1, adminListFilter.CurrentPageIndex);
+        }
+
+        [Test]
+        public void WhenCallIndexWithInValidAdminListRecordPerPage_ThenThrowDivideByZeroException()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 0;
+            Assert.ThrowsAsync(Is.TypeOf<DivideByZeroException>(),
+                         async delegate { await _controller.Index(1, null, null); });
+        }
+
+        [Test]
+        public async Task WhenCallIndex_ThenIsDeletedShouldDisplayYesAndNoRespectively()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 20;
+            IActionResult result = await _controller.Index(1, null, 2020);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.AreEqual(2, adminListFilter.RadioNavigationalWarningsAdminList.Count);
+            Assert.AreEqual("Yes", adminListFilter.RadioNavigationalWarningsAdminList[0].IsDeleted);
+            Assert.AreEqual("No", adminListFilter.RadioNavigationalWarningsAdminList[1].IsDeleted);
+        }
+
+        [Test]
+        public async Task WhenCallIndexWithContentLengthGreaterThan300Char_ThenWrapTheContent()
+        {
+            _fakeRadioNavigationalWarningConfiguration.Value.AdminListRecordPerPage = 20;
+            IActionResult result = await _controller.Index(1, null, 2024);
+            RadioNavigationalWarningsAdminFilter adminListFilter = (RadioNavigationalWarningsAdminFilter)((ViewResult)result).Model;
+            Assert.IsTrue(adminListFilter.RadioNavigationalWarningsAdminList[0].Content.Length <= 303);
```

Comment:
As per test content lenght > 300 gets wraped.
```suggestion
            Assert.IsTrue(adminListFilter.RadioNavigationalWarningsAdminList[0].Content.Length > 303);
```

Reply:
Here I am checking if the actual content in DB is greater than 300 then we are wrapping the content to 300 char with three-dot, that's why I check content length should be less than 303. Hope this will fine ?

Reply:
Thanks for the clarification.